# Chatham_bird_infectomes
Supplementary code and data for Chatham Island bird infectome paper.


<b>Files</b>

Supplementary_data_S1.xlsx - Extended data detailing all the viral polymerases found, accession numbers, closest BLAST hits, sequences (nucleotide and amino acid), other non-viral proteins identified, and PA of non-viral pathogens in the assembled transcriptomes

--

Black_Robin_Viromes.Rmd - R code used to generate bird virome results 

CCmetagen_analysis.Rmd - R code used to generate bird (non-viral) microbiome results

--

NMDS_full_BR_standardised_viral_abundances_condensed.csv - standardised viral abundances per individual sequencing library + metadata

PA_heatmap_BR_raw_viral_abundances.csv - full (extended) raw viral abundances for presence/absence of viruses in samples

NMDS_heatmap_merge_BR_standardised_viral_abundances_condensed_merged.csv - standardised viral abundances by host species and age group for heatmap and NMDS/beta diversity

avian_viromes_RG_condensed.csv - used to produce bipartite network of hosts and viral families

BR_non_vert_viromes_standardised_CONTROLv2.csv - standardised abundances of non-vertebrate viral families for beta diversity non-host virome control 

--

Bird_family_table_filtered_genus.csv - CCmetagen outputs filtered at microbial genus level

CCmetagen_NMDS.csv - filtered CCmetagen data used to perform NMDS analysis of avian (non-viral) microbiomes

ccmetagen_genera_of_interest.csv - logged abundances of microbial genera of interest




